diff --git a/src/application.cpp b/src/application.cpp
index 6b45b2b..bc569c2 100644
--- a/src/application.cpp
+++ b/src/application.cpp
@@ -65,6 +65,7 @@ static QApplication *createApplication(int &argc, char **argv, const QString &ap
 
     QQuickStyle::setStyle(QStringLiteral("org.kde.desktop"));
     QQuickStyle::setFallbackStyle(QStringLiteral("Fusion"));
+    QIcon::setThemeName(QStringLiteral("breeze"));
 
     QApplication *app = new QApplication(argc, argv);
     return app;
