diff --git a/src/main.cpp b/src/main.cpp
index 8dac90a..a2b7c8f 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -35,10 +35,12 @@ int main(int argc, char *argv[])
     QApplication::setOrganizationName("georgefb");
     QApplication::setOrganizationDomain("georgefb.com");
     QApplication::setWindowIcon(QIcon::fromTheme("com.georgefb.haruna"));
-    QQuickStyle::setFallbackStyle(QStringLiteral("org.kde.desktop"));
 
     QApplication app(argc, argv);
 
+    QQuickStyle::setStyle(QStringLiteral("org.kde.desktop"));
+    QQuickStyle::setFallbackStyle(QStringLiteral("fusion"));
+
     KAboutData aboutData(
                 QStringLiteral("haruna"),
                 i18n("Haruna Video Player"),
